src/scripts/profiling/run_profiling.py

```diff
@@ -1,53 +1,232 @@
+    if session is None:
+        return dict()
+
+    profiler_stats = {
+        "start_time": session.start_time,
+        "duration": session.duration,
+        "cpu_time": session.cpu_time,
+    }
+
+    return profiler_stats
+
+
+def record_run_statistics(
+    output_file: str,
+    html_output_file: str = None,
+    profiling_session: Session = None,
+    completed_sim: Simulation = None,
+    disk_usage: Dict[str, Union[int, float]] = None,
+    **additional_stats: str,
+) -> None:
+    """
+    Organise all statistics to be collected from the profiling run into a single dict,
+    which can then be dumped to a JSON file.
+
+    :param output_file: JSON file / path to write to.
+    :param html_output_file: The name of the output HTML file from the profiling run,
+     if it was produced.
+    :param profiled_session: The Session object representing the profiling session.
+    :param completed_sim: The end-state of the simulation.
+    :param disk_usage: Usage stats for the disk I/O operations during the profiling run.
+    :param additional_stats: Dict of any additional information passed by the user that
+     should be recorded.
+    """
+    # Record statistics as [key, value] pairs
+    stats_dict = dict()
+    # Start with user-defined additional stats, if they exist
+    if additional_stats is not None:
+        stats_dict.update(additional_stats)
+
+    # If we wrote a HTML output, include this in the additional statistics to write.
+    # Warn the user if they have overwritten the reserved keyword,
+    # then overwrite the value provided anyway.
+    if html_output_file is not None:
+        if "html_output" in additional_stats.keys():
+            warnings.warn(
+                f"User-provided statistic for 'html_output' was provided: "
+                "this is being overwritten with the path to the HTML output.\n"
+                f"\tWas        : {additional_stats['html_output']}"
+                f"\tReplaced by: {html_output_file}"
+            )
+        stats_dict["html_output"] = html_output_file
+
+    # Fetch statistics from the profiled session itself
+    stats_dict.update(record_profiling_session_statistics(profiling_session))
+
+    # Fetch statistics from end end-state of the simulation
+    stats_dict.update(record_simulation_statistics(completed_sim))
+
+    # Fetch disk I/O statistics
+    stats_dict.update(record_disk_statistics(disk_usage))
+
+    # Having computed all statistics, save the file
+    with open(output_file, "w") as f:
+        json.dump(stats_dict, f, indent=2)
+    return
 
 
 def run_profiling(
     output_dir: Path = PROFILING_RESULTS,
     output_name: Path = None,
-    write_pyis: bool = True,
     write_html: bool = False,
-    write_json: bool = False,
     interval: float = 1e-1,
+    **additional_stats: str,
 ) -> None:
+    """
+    Uses pyinstrument to profile the scale_run simulation,
+    writing the output in the requested formats.
+    """
     # Suppress "ignore" warnings
```

Comment:
```suggestion
```
Comment is a bit misleading I think - the `"ignore"` argument to `warnings.filterwarnings` is the `action` to take not a specification of the warnings to filter for (so the call just specifies to ignore all warnings).